packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -588,33 +598,53 @@ export class ApexDebug extends LoggingDebugSession {
     response: DebugProtocol.SetBreakpointsResponse,
     args: DebugProtocol.SetBreakpointsArguments
   ): Promise<void> {
-    if (args.source && args.source.path && args.lines) {
-      const uri = this.convertClientPathToDebugger(args.source.path);
-      const validBreakpoints = await this.myBreakpointService.reconcileBreakpoints(
-        this.sfdxProject,
-        uri,
-        this.mySessionService.getSessionId(),
-        args.lines.map(line => this.convertClientLineToDebugger(line))
-      );
-      const processedBreakpoints: DebugProtocol.Breakpoint[] = [];
-      validBreakpoints.forEach(validBreakpoint => {
-        processedBreakpoints.push({
-          verified: true,
-          source: args.source,
-          line: this.convertDebuggerLineToClient(validBreakpoint)
-        });
-      });
-      args.lines.forEach(lineArg => {
-        if (!validBreakpoints.has(lineArg)) {
-          processedBreakpoints.push({
-            verified: false,
-            source: args.source,
-            line: this.convertDebuggerLineToClient(lineArg)
+    response = await this.lock.acquire(
```

Comment:
Lock and process the breakpoint request for one file.

Reply:
What happens if we have an error, does it automatically release the lock? Or do we need to have to do something similar to http://www.javapractices.com/topic/TopicAction.do?Id=253

Reply:
The lock will be released when there's an error in the locked request.